test/jdk/javax/swing/JSplitPane/TestSplitPaneEnableTest.java

```diff
@@ -0,0 +1,147 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 5021949
+ * @key headful
+ * @summary  Verifies JSplitPane setEnabled(false) disables one touch expandable clicks
+ * @run main TestSplitPaneEnableTest
+ */
+
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JSplitPane;
+import javax.swing.plaf.basic.BasicSplitPaneDivider;
+import javax.swing.plaf.basic.BasicSplitPaneUI;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class TestSplitPaneEnableTest {
+    private static JButton leftOneTouchButton;
+    private static JButton rightOneTouchButton;
+    private static JFrame frame;
+    private static JSplitPane jsp;
+    private static volatile Point loc;
+    private static volatile int prevDivLoc;
+    private static volatile int curDivLoc;
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF : " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    frame = new JFrame("SplitPaneTest");
+                    jsp = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT,
+                                         new JButton("Left"),
+                                         new JButton("Right"));
+
+                    frame.getContentPane().add(jsp);
+                    jsp.setUI(new TestSplitPaneUI());
+                    jsp.setOneTouchExpandable(true);
+
+                    jsp.setEnabled(false);
+
+                    frame.setSize(400, 200);
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    loc = leftOneTouchButton.getLocationOnScreen();
```

Comment:
Ran the test on ubuntu 22.04, test failed with `IllegalComponentStateException: component must be showing on the screen to determine its location` for GTK L&F.
There is no such touch button available for JSplitPane divider.
May skip testing for GTK.

![JSplitPane_GTK](https://github.com/openjdk/jdk/assets/107542245/428b3f1e-80a0-4cce-8b84-06addc66b135)